cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1077,6 +1094,18 @@ auto build_chunk_dictionaries(hostdevice_2dvector<gpu::EncColumnChunk>& chunks,
       auto dict_enc_size = ck.uniq_data_size + rle_byte_size;
       if (ck.plain_data_size <= dict_enc_size) { return {false, 0}; }
 
+      // don't use dictionary if it gets too large for the given compression codec.
+      // but in the case where the dictionary encoding is giving large savings, prefer
+      // that over compression.
+      if (dict_policy == dictionary_policy::DEFAULT) {
+        constexpr int DICT_THRESHOLD = 4;  // FIXME: need a defensible value here
```

Comment:
Any ideas on how to pick a value for this?

Reply:
A good ballpark estimate of the "average" compression achieved?  If the dictionary is doing well, you'd think the compressor would too, but then the dictionary is over an entire column chunk, while the compressor just gets individual pages.  Also, I added this before adding the `max_dictionary_size` param to the writer, so maybe stop trying to be clever here and just go back to a hard cutoff...if the dictionary is too big don't use it.

It would be nice if we already knew the compressed size before deciding (which I think is how compression is decided later on...aren't all pages compressed, and then compression kept only when the compressed size is smaller?). But then we'd have to encode each page twice to (once with plain encoding and once with dict) to do that comparison, so likely not worth it.

Reply:
I think this also made more sense when the max was 64KB.  With snappy unconstrained and Zstd good up to 16MB now, maybe it's better to remove this test.

Reply:
Even though this was the original intent of the PR, I'm going to punt for now because I think the other changes are more important to get Zstd compression working with nvcomp 2.5.  Without this, one of my tests still throws a memory error, but I can work around that by either setting dictionary policy to `ALWAYS` or bumping up the max dictionary size above the default 1MB.  I do want to revisit this though.